src/core/main.c

```diff
@@ -1480,6 +1478,35 @@ int main(int argc, char *argv[]) {
                         (void) write_string_file("/proc/sys/kernel/core_pattern", "|/bin/false", 0);
         }
 
+        /* Try to figure out if we can use colors with the console. No
+         * need to do that for user instances since they never log
+         * into the console. */
+        if (arg_system) {
+                /* When started as PID1, the kernel uses /dev/console
+                 * for our stdios and uses TERM=linux whatever the
+                 * backend device used by the console. We try to make
+                 * a better guess here since some consoles might not
+                 * have support for color mode for example.
+                 *
+                 * However if TERM was configured through the kernel
+                 * command line then leave it alone.
+                 *
+                 * We expect TERM to be set correctly if run inside a
+                 * container. */
+                if (detect_container() <= 0) {
+                        _cleanup_free_ char *term = NULL;
+
+                        if (!get_proc_cmdline_key("TERM=", &term)) {
+                                term = strdup(default_term_for_tty("/dev/console") + 5);
```

Comment:
If get_proc_cmdline_key() returns false nothing is allocated by this function.

strdup() is used so whatever the returned value of get_proc_cmdline_key(), "term" needs to be freed.